packages/react-slider/src/components/Slider/Slider.test.tsx

```diff
@@ -80,24 +80,59 @@ describe('Slider', () => {
     expect(onChange.mock.calls[0][1]).toEqual({ value: 0 });
   });
 
+  it('slides to the correct position when dragged in-between steps', () => {
+    const onChange = jest.fn();
+    const wrapper: ReactWrapper = mount(
+      <Slider
+        step={10}
+        thumbWrapper={{ className: 'thumb-wrapper' }}
+        activeRail={{ className: 'active-rail' }}
+        onChange={onChange}
+      />,
+    );
+
+    const activeRail = wrapper.find('.active-rail');
+
+    activeRail.getDOMNode().getBoundingClientRect = () =>
+      ({ left: 0, top: 0, right: 100, bottom: 40, width: 100, height: 40 } as DOMRect);
+
+    wrapper.simulate('pointerdown', { type: 'pointermove', clientX: 45, clientY: 0 });
+    expect(onChange).toBeCalledTimes(1);
+    expect(onChange.mock.calls[0][1]).toEqual({ value: 50 });
+    expect(wrapper.find('.thumb-wrapper').props().style?.transform).toEqual('translateX(45%)');
+
+    wrapper.simulate('pointerdown', { type: 'pointermove', clientX: 25, clientY: 0 });
```

Comment:
Maybe for the second run try 24 instead of 25 so that it slides back to 20 instead of 30 and you check both rounding up and rounding down scenarios in the same test.